app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1398,20 +1423,49 @@ class BrowserTabViewModel(
         }
     }
 
-    fun onAddFavoriteMenuClicked() {
+    private suspend fun editSiteBookmark(url: String) {
+        withContext(dispatchers.io()) {
+            bookmarksState.value?.firstOrNull { it.url == url }
+        }?.let {
+            withContext(dispatchers.main()) {
+                command.value = ShowEditSavedSiteDialog(it)
+            }
+        }
+    }
+
+    fun onFavoriteMenuClicked() {
         val url = url ?: return
-        val title = title ?: ""
 
         val buttonHighlighted = currentBrowserViewState().addFavorite.isHighlighted()
-        pixel.fire(
-            AppPixelName.MENU_ACTION_ADD_FAVORITE_PRESSED.pixelName,
-            mapOf(FAVORITE_MENU_ITEM_STATE to buttonHighlighted.toString())
-        )
+        if (currentBrowserViewState().isFavorite) {
+            pixel.fire(AppPixelName.MENU_ACTION_REMOVE_FAVORITE_PRESSED.pixelName)
+            removeFavoriteSite(url)
+        } else {
+            pixel.fire(
+                AppPixelName.MENU_ACTION_ADD_FAVORITE_PRESSED.pixelName,
+                mapOf(FAVORITE_MENU_ITEM_STATE to buttonHighlighted.toString())
+            )
+            saveFavoriteSite(url, title ?: "")
+        }
+    }
 
-        saveSiteAsFavorite(url, title)
+    private fun removeFavoriteSite(url: String) {
+        viewModelScope.launch {
+            withContext(dispatchers.io()) {
+                currentCtaViewState().favorites.firstOrNull {
```

Comment:
same here, the call to `Repository `already triggers a call through the flow, and the `ViewState `should be updated